include/h2o.h

```diff
@@ -86,6 +86,8 @@ extern "C" {
 #define H2O_DEFAULT_PROXY_WEBSOCKET_TIMEOUT_IN_SECS 300
 #define H2O_DEFAULT_PROXY_WEBSOCKET_TIMEOUT (H2O_DEFAULT_PROXY_WEBSOCKET_TIMEOUT_IN_SECS * 1000)
 
+#define H2O_BUILD_ASSERT(condition) ((void)sizeof(char[1 - 2*!!(condition)]))
```

Comment:
I have three concerns.

* please move the definition to `h2o/memory.h`, since that is the file tha currently has these kind of very basic macros
* shouldn't the size of the array be calculated as `2*!!(condition) - 1` instead?
 * current formula evaluates to 1 if `condition` is 0, and to 0 if `condition` is 1
*  it might be better to replace `condition` with `!__builtin_constant_p(condition) || (condition)`, so that 
`H2O_BUILD_ASSERT` gets silently skipped in case the condition is a non-constant